simple-js-examples/simple-fota-example.js

```diff
@@ -0,0 +1,545 @@
+		console.log(GREEN, 'Connecting to "', url, '"');
+		self.client.connect(url, function connected(error, reply) {
+			if (!error) {
+				resolve(self);
+			} else {
+				reject(error);
+			}
+		});
+	});
+};
+
+EdgePTExample.prototype.disconnect = async function () {
+	let self = this;
+	return new Promise((resolve, reject) => {
+		console.log(GREEN, "Disconnecting from Edge.");
+		self.client.disconnect((error, response) => {
+			if (!error) {
+				resolve(response);
+			} else {
+				reject(error);
+			}
+		});
+	});
+};
+
+EdgePTExample.prototype.exposeWriteMethod = function() {
+		let self = this;
+		self.client.expose('write', (params, response) => {
+				let value = new Buffer.from(params.value, 'base64').readDoubleBE();
+				let resourcePath = params.uri.objectId + '/' + params.uri.objectInstanceId
+						+ '/' + params.uri.resourceId;
+				let deviceId = params.uri.deviceId;
+				let operation = '';
+				if (params.operation === OPERATIONS.WRITE) {
+						operation = 'write';
+				} else if (params.operation === OPERATIONS.EXECUTE) {
+						operation = 'execute';
+				} else {
+						operation = 'unknown';
+				}
+
+				received = {
+						deviceId: deviceId,
+						resourcePath: resourcePath,
+						operation: operation,
+						value: value
+				}
+				console.log(GREEN, 'Received a write method with data:');
+				console.log(received);
+				console.log(GREEN, 'The raw received JSONRPC 2.0 params:');
+				console.log(params);
+
+				/* Always respond back to Edge, it is expecting
+				 * a success response to finish the write/execute action.
+				 * If an error is returned the value write is discarded
+				 * also in the Edge Core.
+				 */
+				response(/* no error */ null, /* success */ 'ok');
+		});
+};
+
+EdgePTExample.prototype.registerProtocolTranslator = async function () {
+	let self = this;
+	return new Promise((resolve, reject) => {
+		let timeout = setTimeout(() => {
+			reject("Timeout");
+		}, TIMEOUT);
+
+		self.client.send(
+			"protocol_translator_register",
+			{ name: self.name },
+			function (error, response) {
+				clearTimeout(timeout);
+				if (!error) {
+					// Connection ok. Set up to listen for write calls
+					// from Edge Core.
+				self.exposeVendorandClass();
+				self.exposeWriteMethod();
+					resolve(response);
+				} else {
+					reject(error);
+				}
+			}
+		);
+	});
+};
+
+// update 10252/0/3 with  and 10252/0/2 after completion of update
+
+EdgePTExample.prototype._createDeviceParams = function (
+	deviceId,
+	temperatureValue,
+	setPointValue,
+	version
+) {
+	// Values are always Base64 encoded strings.
+	let temperature = Buffer.allocUnsafe(4);
+	temperature.writeFloatBE(temperatureValue);
+	temperature = temperature.toString("base64");
+	let setPoint = Buffer.allocUnsafe(4);
+	setPoint.writeFloatBE(setPointValue);
+	setPoint = setPoint.toString("base64");
+	let data = "0";
+	let buff = new Buffer.from(data);
+	let base64data = buff.toString("base64");
+
+	let prot_version = 4;
```

Comment:
It would be good if there's a comment or indication that these values are just for example or they have to be exactly these values for fota to work.
Like protocol version being 4 here.

Reply:
Comment added in the latest commit